tests/unit/test_matrix.py

```diff
@@ -244,19 +245,83 @@ def it_knows_how_to_assemble_its_blocks(self, request):
             [bs._subtotal_rows, bs._intersections],
         ]
 
+    @pytest.mark.parametrize(
+        ("nrows", "column_subtotals", "expected"),
+        (
+            (3, [], np.empty((3, 0), dtype=np.float64)),
+            (3, [1, 2], np.array([[1, 1], [2, 2], [3, 3]])),
+        ),
+    )
+    def it_knows_its_subtotal_columns(
+        self,
+        _column_subtotals_prop_,
+        _nrows_prop_,
+        _subtotal_column_,
+        nrows,
+        column_subtotals,
+        expected,
+    ):
+        _column_subtotals_prop_.return_value = column_subtotals
```

Comment:
The column-subtotals returned by `._column_subtotals` is a `cr.cube.dimension._Subtotals` object. It's probably okay to mock those with a tuple of `_Subtotal` instance mocks, but I don't think a list is going to get it done.

Reply:
This one might be good to use two tests for, at least to get started. You can always combine them later if they're similar enough, but starting with the empty case is probably easiest and will allow you to make progress. These can be something like:

```
it_produces_an_empty_column_subtotals_matrix_when_there_are_no_column_subtotals()

# --- and ---

it _produces_an_nrows_x_n_col_subtotals_matrix_when_there_are_column_subtotals()
```

One advantage of that approach is you don't have to squeeze your fixtures into a compact set of parameters, so the fixture is less abstract and easier to read.

Reply:
i think i figured it out ok now..